web-app-router/scenes/Portal/Teams/TeamId/Apps/AppId/Actions/ActionId/ProofDebuging/Debugger/index.tsx

```diff
@@ -0,0 +1,254 @@
+    watch,
+    setValue,
+    setError,
+    formState: { errors, isDirty, isValid },
+  } = useForm<TestProofFormValues>({
+    resolver: yupResolver(testProofSchema),
+    mode: "onChange",
+  });
+
+  const signal = watch("signal");
+  const verificationResponse = watch("verification_response");
+
+  useEffect(() => {
+    if (isDirty) {
+      setStatus(Status.IDLE);
+      setMessage(messages.IDLE);
+    }
+  }, [signal, verificationResponse, isDirty]);
+
+  const submit = async (formData: TestProofFormValues) => {
+    try {
+      setStatus(Status.IDLE);
+      const verification = JSON5.parse(formData.verification_response ?? "");
+      const res = await fetch("/api/v1/debugger", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          ...verification,
+          app_id: appID,
+          action: action?.action,
+          signal: formData.signal,
+          is_staging: action.app?.is_staging,
+        }),
+      });
+
+      const response = await res.json();
+      // NOTE: missing attr in response
+      if (response.code === "invalid") {
+        setStatus(Status.ERROR);
+        setMessage(messages.MISSING_ATTR(response.attribute));
+        return;
+      }
+
+      // NOTE: other errors
+      if (res.status !== 200) {
+        setStatus(Status.ERROR);
+        setMessage(messages.NOT_VERIFIED);
+        // posthog.capture("debugger-verify-failed");
+        return;
+      }
+
+      // NOTE: success mined
+      setStatus(Status.SUCCESS);
+      if (response.status === "on-chain") {
+        setMessage(messages.SUCCESS_ONCHAIN);
+        // posthog.capture("debugger-verify-success", { environment: "on-chain" });
+      } else {
+        setMessage(messages.SUCCESS_PENDING);
+        // posthog.capture("debugger-verify-success", { environment: "cloud" });
+      }
+    } catch (err) {
+      setStatus(Status.ERROR);
+      setMessage(messages.REQUEST_ERROR);
+    }
+  };
+
+  const formatCode = useCallback(
+    async (data: TestProofFormValues) => {
+      try {
+        const json = JSON5.parse(data.verification_response ?? "");
+        setValue("verification_response", JSON5.stringify(json, null, 2));
+      } catch (e) {
+        setError("verification_response", {
+          type: "manual",
+          message: "Unable to auto format the response",
+        });
+      }
+    },
+    [setValue, setError]
+  );
+  return (
+    <form
+      className="w-full grid-cols-2 grid items-start justify-between gap-x-32"
+      onSubmit={handleSubmit(submit)}
+    >
+      <div className="w-full grid gap-y-6">
+        <h1 className="text-lg font-[550]">Output parameters</h1>
+        <div className="grid gap-y-8">
+          <Input
+            register={register("signal")}
+            helperText="Enter the signal as passed to IDKit"
+            label="Signal"
+            placeholder="my_signal"
+          />
+          <TextArea
```

Comment:
@andy-t-wang What do you think about to try something like [embedded editor](https://www.npmjs.com/package/@monaco-editor/react) here

Reply:
I tried it and found it quite a pain to style and match the existing spec. Perhaps something we revisit in the future. For now the format code button should help.